src/main/scala/com/teracode/beacons/routing/LocationService.scala

```diff
@@ -50,46 +67,70 @@ trait LocationService extends HttpService {
+    new ApiResponse(code = 404, message = "Location for the given ID not found.")
   ))
-  def getRoute = get {
+  def deleteRoute = delete {
     path(LocationsPath / JavaUUID) { locationId =>
-      onSuccess(locationESStorage.get(locationId)) {
-        case Some(location)   => complete(location)
-        case None             => complete(NotFound)
+      onSuccess(storage.delete(locationId)) {
+        case true   => complete(NoContent)
+        case false  => complete(NotFound)
       }
     }
   }
 
-  @ApiOperation(value = "Deletes a Location", nickname = "deleteLocation", httpMethod = "DELETE")
+  @ApiOperation(value = "Search Location by description", nickname = "DescriptionLocationSearch", position = 4, httpMethod = "POST", consumes = "application/json", produces = "application/json, application/xml")
   @ApiImplicitParams(Array(
-    new ApiImplicitParam(name = "id", value = "Location Id to delete", required = true, dataType = "JavaUUID", paramType = "path")
+    new ApiImplicitParam(name = "Description", value = "Description", required = true, dataType = "String", paramType = "body")
   ))
-  @ApiResponses(Array(
-    new ApiResponse(code = 404, message = "Location for the given ID not found.")
+  @Path("/search")
+  def DescriptionSearch = post {
+    path(LocationsPath / "search") {
+      decompressRequest() {
+        entity(as[String]) { str =>
+          onSuccess(storage.search(str)) { result =>
+            complete(result)
+          }
+        }
+      }
+    }
+  }
+
+  @ApiOperation(value = "Search Location by Signals", nickname = "SignalsLocationSearch", position = 5, httpMethod = "POST", consumes = "application/json", produces = "application/json, application/xml")
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "signals", value = "List of Beacon", required = true, dataType = "SignalSearch", paramType = "body")
   ))
-  def deleteRoute = delete {
-    path(LocationsPath / JavaUUID) { locationId =>
-      onSuccess(locationESStorage.delete(locationId)) {
-        case true   => complete(NoContent)
-        case false  => complete(NotFound)
+  @Path("/signalsearch")
+  def SignalSearchRoute = post {
+    path(LocationsPath / "signalsearch") {
+      decompressRequest() {
+        entity(as[SignalSearch]) { signalSearch =>
+          onSuccess(storage.search(signalSearch)) { result =>
+            complete(result)
+          }
+        }
       }
     }
   }
 
-  @ApiOperation(value = "Searches for a Location", nickname = "searchLocation", httpMethod = "GET", produces = "application/json, application/xml")
+
+  @ApiOperation(value = "Searches for all Location", nickname = "search all Location", position = 6, httpMethod = "GET", produces = "application/json, application/xml")
   def searchRoute = get {
     path(LocationsPath) {
-      onSuccess(locationESStorage.search()) { result =>
+      onSuccess(storage.search()) { result =>
         complete(result)
       }
     }
   }
 
+
```

Comment:
These line are here to make swagger recognize SignalSearch as a type. With this lines, swagger shows SignalSearch as a model with its schema.
It should be removed with the new version of swagger-core according to the issue in the commentary.